services/welding-machine-defect-detection-model-service/app/models/model_loader.py

```
@@ -0,0 +1,41 @@
+import joblib
+import json
+import numpy as np
+from keras.models import load_model
+
+
+def load_scaler(signal_type: str):
+    try:
+        return joblib.load(f"app/models/scalers/welding_scaler_{signal_type}.pkl")
+    except FileNotFoundError:
+        raise FileNotFoundError(
+            f"Scaler file for signal type '{signal_type}' not found")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)